python/pyarrow/tests/test_compute.py

```diff
@@ -2691,16 +2691,28 @@ def test_struct_fields_options():
     b = pa.array(["bar", None, ""])
     c = pa.StructArray.from_arrays([a, b], ["a", "b"])
     arr = pa.StructArray.from_arrays([a, c], ["a", "c"])
-
-    assert pc.struct_field(arr,
-                           indices=[1, 1]) == pa.array(["bar", None, ""])
-    assert pc.struct_field(arr, [1, 1]) == pa.array(["bar", None, ""])
-    assert pc.struct_field(arr, [0]) == pa.array([4, 5, 6], type=pa.int64())
+    assert pc.struct_field(arr, '.c.b', is_dot_path=True) == b
+    assert pc.struct_field(arr, '.a', is_dot_path=True) == a
+    assert pc.struct_field(arr, 'a') == a
+    assert pc.struct_field(arr, indices=[1, 1]) == b
+    assert pc.struct_field(arr, [1, 1]) == b
```

Comment:
I think it should also be possible to pass the names as a list? (`['b', 'c']`)

Reply:
Which would be equivalent to a dot path of `.b.c`, is that right?

Reply:
Yes, and we should probably allow passing an actual FieldRef instance as well (`pc.field(("b", "c"))`)

Personally, I am not sure we necessarily need to support a dotted path here, especially if it requires a keyword (which seems to defeat the purpose of added convenience a bit?)

Reply:
Could do the same assumption `FromDotPath` does and create one if it has a leading `.`, then remove the kwarg?

Reply:
Done in https://github.com/apache/arrow/pull/14495/commits/0ff256726c100e791378534c9a7fdbd8d5c6dc1b and https://github.com/apache/arrow/pull/14495/commits/a4b5ce00dae66fe9a329359700d3db767eb31e20